src/XrdPfc/XrdPfc.cc

```diff
@@ -1097,6 +1180,41 @@ int Cache::Prepare(const char *curl, int oflags, mode_t mode)
    if (m_oss->Stat(i_name.c_str(), &sbuff) == XrdOssOK)
    {
       TRACE(Dump, "Prepare defer open " << f_name);
+
+      std::string icc;
+      if (GetCacheControlXAttr(i_name, icc) > 0) {
+         using namespace nlohmann;
+         json j = json::parse(icc);
+         if (j.contains("ETag") && j.contains("revalidate") && j["revalidate"] == true) {
+            // comapre cinfo xattr etag and the etag from http header response
+            XrdCl::FileSystem fs(url);
+            XrdCl::Buffer queryArgs(500);
+            queryArgs.FromString(curl); // pass parh throug args
+            XrdCl::Buffer *response = nullptr;
+
+            XrdCl::XRootDStatus st = fs.Query(XrdCl::QueryCode::XAttr, queryArgs, response);
+
+            if (st.IsOK())
+            {
+               std::string etag = response->ToString();
+               bool etagValid = (etag == j["ETag"]);
+               TRACE(Info, "Prepare " << f_name << ", ETag valid res: " << etagValid);
+               return etagValid;
+            }
+            else
+            {
+               TRACE(Error, "Prepare() XrdCl::FileSystem::Query failed " << f_name.c_str());
```

Comment:
this should not be an error ... xrootd servers might not report this at all. we only care for files that we got the CC-header for on initial open.

Reply:
This is already in the xattr condition